commands/dns_cli.py

```diff
@@ -92,19 +101,27 @@ def create_cert(client, domain_client, domain, base_len):
         response = client.describe_certificate(CertificateArn=arn)
 
     cert_record = response["Certificate"]["DomainValidationOptions"][0]["ResourceRecord"]
-    response = domain_client.list_hosted_zones_by_name()
 
+    click.secho(f"Looking fo ID of domain {domain_to_create}...", fg="yellow")
+    domain_id = False
+    response = domain_client.list_hosted_zones_by_name()
     for hz in response["HostedZones"]:
         if hz["Name"] == domain_to_create:
             domain_id = hz["Id"]
             break
+    if not domain_id:
+        # Will got here more than once during manual testing, it might be a race condition we need to handle better
+        click.secho(f"Unable to find Domain ID for {domain_to_create} in the hosted zones", fg="red", bold=True)
+        exit(1)
```

Comment:
Previously it could carry on past here without any clue that the result of it's labours were not going to work 😆